plugins/package-managers/node/src/funTest/kotlin/yarn/YarnFunTest.kt

```diff
@@ -51,6 +52,16 @@ class YarnFunTest : StringSpec({
         result.toYaml() should matchExpectedResult(expectedResultFile, definitionFile)
     }
 
+    "Resolve dependencies for a project with an invalid package.json file correctly" {
+        val definitionFile = getAssetFile("projects/synthetic/yarn/invalid-package-json/package.json")
+        val expectedResultFile = getAssetFile("projects/synthetic/yarn/invalid-package-json-expected-output.yml")
```

Comment:
The test name can be misleading IMO, how about this?

"Resolve dependencies for a project which installs a module with an invalid unused package.json"